### PR TITLE
Add basic forking cheatcodes (continuation of #259)

### PR DESCRIPTION
**Pull request by samalws-tob**
Continuation of #259 (@arcz suggested I open a new PR rather than iterate on his)

## Description
Basic implementation of forking: https://book.getfoundry.sh/forge/fork-testing

## Checklist

- [X] tested locally
- [X] added automated tests
- [X] updated the docs
- [X] updated the changelog

### CODE REVIEW

File: src/EVM.hs [hunk 1]
```diff
@@ -161,7 +148,26 @@ makeVm o = do
...
+      { coinbase = o.coinbase
+      , timestamp = o.timestamp
+      , number = o.number
+      , prevRandao = o.prevRandao
+      , maxCodeSize = o.maxCodeSize
+      , gaslimit = o.blockGaslimit
+      , baseFee = o.baseFee
+      , schedule = o.schedule
+      }
+    cache = Cache mempty mempty
```
Comment (msooseth) [inline src/EVM.hs:170]: Is this refactoring necessary? I don't want to change code unless we need to or we should... I am not amazing at Haskell refactoring but I'm not sure this is necessary? I may be wrong...

**Comment by arcz** [comment]
@samalws-tob could you rebase? there were a few changes since then

**Comment by samalws-tob** [comment]
@arcz done

File: CHANGELOG.md [hunk 2]
```diff
@@ -28,6 +28,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - New concrete fuzzer that can be controlled via `--num-cex-fuzz`
 - Partial support for dynamic jumps when the jump destination can be computed
   given already available information
+- Two basic forking cheatcodes: `createFork` and `selectFork`
```
Comment (msooseth) [inline CHANGELOG.md:31]:
```suggestion
- Added two forking cheatcodes: `createFork` and `selectFork`
```

**Comment by msooseth** [comment]
Looks really nice actually, thank you so much! Do you think it'd make sense to also implement `activeFork` if we are at it? I am thinking it'd be like 5 lines of code, perhaps best to add it to this PR, rather than having to get back into the same mindspace a few months later and do it then, with more effort? What do you think?

**Comment by samalws-tob** [comment]
Just added `activeFork`

**Comment by msooseth** [comment]
Thanks, merged! Sorry for being late :(

**Pull request merged**